js/index.js

```diff
@@ -1,35 +1,39 @@
 // ITERATION 1
 
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
-
-  //... your code goes here
+  const price = product.querySelector(".price span").innerText;
+  const quantity = product.querySelector(".quantity input").value;
+  const subTotalPrice = quantity * price;
+  const subTotal = product.querySelector(".subtotal span");
+  subTotal.innerText = subTotalPrice;
+  return subTotalPrice;
 }
 
-function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
 
-  // ITERATION 2
-  //... your code goes here
+// ITERATION 2-3
 
-  // ITERATION 3
-  //... your code goes here
+function calculateAll() {
+  let priceTotal = 0;
+  const priceTotalInnerText = document.querySelector("#total-value span");
+  const productArray = [...document.getElementsByClassName("product")];
+  for (let i=0; i < productArray.length; i++){
+      updateSubtotal(productArray[i]);
+      priceTotal += updateSubtotal(productArray[i]);
+  }
```

Comment:
the above is perfect for a reduce... and nobody did it :( 
```js
const priceTotal = productArray.reduce((total, product) => total + updateSubtotal(product)), 0);
```

Reply:
also with your solution you are calling updateSubtotal twice, wasting precious computing resources completely unnecessarily.

Reply:
Sorry about not reducing! :D